test/udp_tracker.cpp

```diff
@@ -171,6 +173,7 @@ struct udp_tracker
 	{
 		m_socket.cancel();
 		m_socket.close();
+		m_ios.stop();
```

Comment:
thanks for looking into this. I feel a little bit worried about using the stopped state of the ios to signal the thread. In the scenario that the main thread calls stop() on the io_service immediately after the thread passes its own check, and issues another outstanding operation. will that operation fail or will the io_service automatically start again? I imagine it would immediately fail, but I'm not 100% certain. an alternative to this would be to have a member variable bool m_abort, post a message to the thread to set this to true and check against that instead.

Actually, why does m_ios.run() return? if the thread issues another async operation from the handler of the previous one, run() would stay blocked, right?